### PR TITLE
Add support for multiple LZ4 frames.

### PR DESCRIPTION
**Pull request by xrvdg**
Multiple LZ4 frames are useful whenever you want to chunk a ByteString into parts that are individually decompressable.

### CODE REVIEW

File: src/Codec/Compression/LZ4/Conduit.hsc [hunk 1]
```diff
@@ -469,41 +503,44 @@ compressWithOutBufferSize bufferSize =
...
+
+          if remainingCapacity >= footerSize
+            then do
+              go remainingCapacity
+            else do
+              -- Footer doesn't fit: Yield buffer, put footer into now-free buffer
+              yieldOutBuf (outBufferSize - remainingCapacity)
+              go outBufferSize
 
     let loop remainingCapacity = do
```
Comment (crtschin) [inline src/Codec/Compression/LZ4/Conduit.hsc:533]: I'm not super convinced this actually does what we hope it does. We're reusing the same compression context across frames. From what I understand in reading about LZ77, wouldn't you then be able to encounter tokens, whose uncompressed versions occurred in an earlier frame?

I believe the safest alternative is to implement `compressWithOutBufferSizeMultiFrame` by calling/restarting `compressWithOutBufferSize` on each `Flush`.

**Review by crtschin — COMMENTED** [review]
I have some concerns. Could you also add some unit tests that test the multiframe compression function.

Api-wise, it might also be useful to already add a corresponding `decompressMultiFrame` function that handles multiple frames in a flat `Conduit ByteString ByteString` stream and automatically handles headers/footers and verifies checksums by looking for the magic numbers and ending marks.

File: src/Codec/Compression/LZ4/Conduit.hsc (hunk 1 above)
Comment (xrvdg) [reply src/Codec/Compression/LZ4/Conduit.hsc:533]: The struct for bookkeeping the compression context is being reused, not the actual compression context itself.  From the LZ4 manual:
> A successful call to LZ4F_compressEnd() makes `cctx` available again for another compression task.

I've verified this by reading the source for `LZ4F_compressBegin` and `LZ4F_compressEnd`. Will add a comment for this.

**Comment by xrvdg** [comment]
> I have some concerns. Could you also add some unit tests that test the multiframe compression function.

These have now been added. 

> Api-wise, it might also be useful to already add a corresponding decompressMultiFrame function that handles multiple frames in a flat Conduit ByteString ByteString stream and automatically handles headers/footers ... 

I've modified `decompress` such that it can handle both single and multiple LZ4 frames.

**Review by crtschin — APPROVED** [review]
LGTM

**Pull request merged**